<launch>
    <arg name="vehicle_id" default="vehicle_0" />
    <arg name="vehicle_id_leader" default="vehicle_0" />
    <arg name="hardware_id" default="default" />
    <arg name="velocity_pwm_map_path" default="$(find rcv)/assets/$(arg hardware_id).csv" />

    <!-- Common nodes -->
    <node name="$(eval vehicle_id + '_esc_node')" pkg="rcv" type="esc.py" respawn="true" />
    <node name="$(eval vehicle_id + '_servo_node')" pkg="rcv" type="servo.py" respawn="true" />
    <node name="$(eval vehicle_id + '_velocity_node')" pkg="common" type="velocity.py" respawn="true" />
    <node name="$(eval vehicle_id + '_debug_node')" pkg="common" type="debug.py" respawn="true" />

    <param name="VEHICLE_ID" type="string" value="$(arg vehicle_id)" />
    <param name="VEHICLE_ID_LEADER" type="string" value="$(arg vehicle_id_leader)" />
    <param name="VEHICLE_WHEEL_RADIUS_CM" type="double" value="3.25" />

    <!-- GPIOs BOARD -->
    <param name="REFLECTIVE_SENSOR_PIN" type="int" value="37" />

    <!-- GPIOs BCM (i.e. the number after GPIO) -->
    <param name="SERVO_PIN" type="int" value="12" />
    <param name="MOTOR_PIN" type="int" value="13" />

    <!-- Velocity -->
    <param name="VELOCITY_WAIT_TIME_US" type="double" value="2000" />
    <param name="VELOCITY_REFLECTANCE_THRESHOLD_US" type="int" value="200" />
    <param name="VELOCITY_TAPES_PER_ROTATION" type="int" value="6" />
    <param name="VELOCITY_PWM_MAP_PATH" type="string" value="$(arg velocity_pwm_map_path)" />
    <param name="VELOCITY_PWM_MAP_POLYFIT_DEGREE" type="int" value="3" />

    <!-- Publisher periods -->
    <param name="VELOCITY_PUBLISH_PERIOD" type="double" value="0.07" />
    <param name="DEBUG_PUBLISH_PERIOD" type="double" value="0.3" />
    <param name="MESSAGE_QUEUE_SIZE" type="int" value="10" />

    <!-- Motor node -->
    <param name="MAX_FORWARD_MOTOR" type="int" value="770000" />
    <param name="MIN_FORWARD_MOTOR" type="int" value="715000" />
    <param name="MAX_REVERSE_MOTOR" type="int" value="585000" />
    <param name="IDLE_MOTOR" type="int" value="660000" />
    <param name="PWM_FREQUENCY_MOTOR" type="int" value="500" />
    <param name="MAX_RIGHT_ANGLE" type="int" value="870000" />
    <param name="MAX_LEFT_ANGLE" type="int" value="270000" />
    <param name="ZERO_ANGLE" type="int" value="560000" />
</launch>
